src/native/managed/cdacreader/Microsoft.Diagnostics.DataContractReader.Contracts/TargetStream.cs

```diff
@@ -0,0 +1,58 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+
+namespace Microsoft.Diagnostics.DataContractReader;
+
+internal class TargetStream(Target target, ulong startPosition, long size) : Stream
+{
+    private readonly ulong _startPosition = startPosition;
+    private long _offset;
+    private readonly long _size = size;
+    private readonly Target _target = target;
+
+    private ulong GlobalPosition => _startPosition + (ulong)_offset;
+
+    public override bool CanRead => true;
+
+    public override bool CanSeek => true;
+
+    public override bool CanWrite => false;
+
+    public override long Length => _size;
+
+    public override long Position { get => _offset; set => _offset = value; }
+
+    public override unsafe int Read(byte[] buffer, int offset, int count)
+    {
+        Span<byte> span = buffer;
+        return Read(span.Slice(start: offset, length: count));
+    }
+    public override unsafe int Read(Span<byte> buffer)
+    {
+        _target.ReadBuffer(GlobalPosition, buffer);
+        _offset += buffer.Length;
+        return buffer.Length;
+    }
+    public override long Seek(long offset, SeekOrigin origin)
+    {
+        switch (origin)
+        {
+            case SeekOrigin.Begin:
+                _offset = offset;
+                break;
+            case SeekOrigin.Current:
+                _offset += offset;
+                break;
+            case SeekOrigin.End:
+                throw new NotSupportedException();
+        }
+        return _offset;
+    }
+
+    public override void Flush() => throw new NotImplementedException();
```

Comment:
Based on the Stream docs, this should be a no-op for a read-only stream.

Reply:
Perhaps a debug-only assert might be better.
```suggestion
    public override void Flush() => Debug.Assert(false, "Flush() should not be called on TargetStream.");
```

Reply:
I think I prefer leaving it as a no-op, in theory is allowed to call Flush() according to the spec.